netutils/nist.py

```diff
@@ -0,0 +1,130 @@
+"""Functions building NIST URLs from the os platform values."""
+import re
+import typing as t
+
+
+def get_nist_urls_juniper_junos(os_platform_data: t.Dict[str, t.Any], api_key: str) -> t.List[str]:
+    """Create a list of possible NIST Url strings for JuniperPlatform.
+
+    Args:
+        api_key: NIST-API-KEY - Request here https://nvd.nist.gov/developers/request-an-api-key
+
+    Returns:
+        List of NIST CPE URLs that may contain platform data.
+    """
+    nist_urls = []
+    base_url = f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey={api_key}&addOns=cves&cpeMatchString=cpe:2.3:o:juniper:junos"""
+
+    url_args = {"base_url": base_url, **os_platform_data}
+    url_args["delim_six"] = ":*" * 6
+    url_args["delim_seven"] = ":*" * 7
+
+    if os_platform_data.get("isspecial"):
+        url_args["type"] = url_args.get("type").lower()  # type: ignore
+        # juniper:junos:12.1x47
+        base_ext = f"{url_args['base_url']}:{url_args.get('main')}.{url_args.get('minor')}{url_args.get('type')}{url_args.get('build')}"
```

Comment:
Can we remove the .get's in f-strings? My gut says that if `url_args.get('main')` doesn't exist, you do not want it to return a `None` string anyway.

Reply:
Removed these in latest commit.